isis/src/control/objs/BundleUtilities/BundleObservationSolveSettings.cpp

```diff
@@ -47,38 +47,157 @@ namespace Isis {
+    }
+    else if (ssolve == "ACCELERATIONS") {
+      m_instrumentPositionSolveOption = PositionVelocityAcceleration;
+      m_numberCamPosCoefSolved = 3;
+    }
+    else if (csolve == "ALL"){
+      m_instrumentPositionSolveOption = AllPositionCoefficients;
+      m_numberCamPosCoefSolved = m_spkSolveDegree + 1;
+    }
+
+    m_positionAprioriSigma.clear();
+
+    // If TWIST is not specified, then a default of YES is used
+    if (scParameterGroup.hasKeyword("TWIST")) {
+      QString parval = (QString)scParameterGroup.findKeyword("TWIST");
+      parval = parval.toUpper();
+      if (parval == "TRUE" || parval == "YES") { // is this necessary ??? i think pvl and toString can handle it...
+        m_solveTwist = true;
+      }
+      else if (parval == "FALSE" || parval == "NO") {
+        m_solveTwist = false;
+      }
+      else {
+        QString msg = "The TWIST parameter must be set to TRUE or FALSE; YES or NO";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+    }
+    // If OVERHERMITE is not specified, then a default of NO is used
+    if (scParameterGroup.hasKeyword("OVERHERMITE")) {
+     QString parval = (QString)scParameterGroup.findKeyword("OVERHERMITE");
+     parval = parval.toUpper();
+     if (parval == "TRUE" || parval == "YES") {
+       m_solvePositionOverHermiteSpline = true;
+       m_positionInterpolationType = SpicePosition::PolyFunctionOverHermiteConstant;
+     }
+     else if (parval == "FALSE" || parval == "NO") {
+       m_solvePositionOverHermiteSpline = false;
+       m_positionInterpolationType = SpicePosition::PolyFunction;
+     }
+     else {
+       QString msg = "The OVERHERMITE parameter must be set to TRUE or FALSE; YES or NO";
+       throw IException(IException::User, msg, _FILEINFO_);
+     }
+    }
+
+    if (csolve != "NONE") {
+     if (scParameterGroup.hasKeyword("CAMERA_ANGLES_SIGMA"))
+       m_anglesAprioriSigma.append((double)(scParameterGroup.findKeyword("CAMERA_ANGLES_SIGMA")));
+     if (scParameterGroup.hasKeyword("CAMERA_ANGULAR_VELOCITY_SIGMA"))
+       m_anglesAprioriSigma.append(
+           (double)(scParameterGroup.findKeyword("CAMERA_ANGULAR_VELOCITY_SIGMA")));
+     if (scParameterGroup.hasKeyword("CAMERA_ANGULAR_ACCELERATION_SIGMA"))
+       m_anglesAprioriSigma.append(
+           (double)(scParameterGroup.findKeyword("CAMERA_ANGULAR_ACCELERATION_SIGMA")));
+    }
+
+    if (ssolve != "NONE") {
+     if (scParameterGroup.hasKeyword("SPACECRAFT_POSITION_SIGMA"))
+       m_positionAprioriSigma.append((double)(scParameterGroup.findKeyword("SPACECRAFT_POSITION_SIGMA")));
+     if (scParameterGroup.hasKeyword("SPACECRAFT_VELOCITY_SIGMA"))
+       m_positionAprioriSigma.append((double)(scParameterGroup.findKeyword("SPACECRAFT_VELOCITY_SIGMA")));
+     if (scParameterGroup.hasKeyword("SPACECRAFT_ACCELERATION_SIGMA"))
+       m_positionAprioriSigma.append((double)(scParameterGroup.findKeyword("SPACECRAFT_ACCELERATION_SIGMA")));
```

Comment:
Same here, look for `ADDITIONAL_SPACECRAFT_POSITION_SIGMAS` and append those to `m_positionAprioriSigma`

Reply:
Not sure how this would look. In the `ADDITIONAL_SPACECRAFT_POSITION_SIGMAS` keyword, would it contain a list of doubles and then I would loop through that list and append them all? I'm also guessing I don't need to append null if this keyword isn't found

Reply:
It would be an array of doubles, you can use `PvlKeyword::size` and `PvlKeyword::operator[](int index)` to loop through the values and append them. It is purely for sigmas past the acceleration.